sys-devel/clang-common/clang-common-16.0.0_pre20220915.ebuild

```diff
@@ -3,22 +3,105 @@
 
 EAPI=8
 
-inherit bash-completion-r1 llvm.org
+inherit bash-completion-r1 llvm.org multilib
 
 DESCRIPTION="Common files shared between multiple slots of clang"
 HOMEPAGE="https://llvm.org/"
 
 LICENSE="Apache-2.0-with-LLVM-exceptions UoI-NCSA"
 SLOT="0"
 KEYWORDS=""
+IUSE="default-compiler-rt default-libcxx default-lld llvm-libunwind"
 
 PDEPEND="
 	sys-devel/clang:*
+	default-compiler-rt? (
+		=sys-libs/compiler-rt-${PV%_*}*
+		llvm-libunwind? ( sys-libs/llvm-libunwind )
+		!llvm-libunwind? ( sys-libs/libunwind )
+	)
+	default-libcxx? ( >=sys-libs/libcxx-${PV} )
+	default-lld? ( sys-devel/lld )
 "
 
 LLVM_COMPONENTS=( clang/utils )
 llvm.org_set_globals
 
+pkg_pretend() {
+	[[ ${CLANG_IGNORE_DEFAULT_RUNTIMES} ]] && return
+
+	local flag missing_flags=()
+	for flag in default-{compiler-rt,libcxx,lld}; do
+		if ! use "${flag}" && has_version "sys-devel/clang[${flag}]"; then
+			missing_flags+=( "${flag}" )
+		fi
+	done
+
+	if [[ ${missing_flags[@]} ]]; then
+		eerror "It seems that you have the following flags set on sys-devel/clang:"
+		eerror
+		eerror "  ${missing_flags[*]}"
+		eerror
+		eerror "The default runtimes are now set via flags on sys-devel/clang-common."
+		eerror "The build is being aborted to prevent breakage.  Please either set"
+		eerror "the respective flags on this ebuild, e.g.:"
+		eerror
+		eerror "  sys-devel/clang-common ${missing_flags[*]}"
+		eerror
+		eerror "or build with CLANG_IGNORE_DEFAULT_RUNTIMES=1."
+		die "Mismatched defaults detected between sys-devel/clang and sys-devel/clang-common"
+	fi
+}
+
+usec() {
+	usex "default-${1}" "" "# "
+}
+
 src_install() {
 	newbashcomp bash-autocomplete.sh clang
+
+	insinto /etc/clang
+	newins - gentoo-runtimes.cfg <<-EOF
+		# This file is initially generated by sys-devel/clang-runtime.
+		# It is used to control the default runtimes using by clang.
+
+		$(usec compiler-rt)-rtlib=compiler-rt
+		$(usec compiler-rt)-unwindlib=libunwind
+		$(usec libcxx)-stdlib=libc++
+		$(usec lld)-fuse-ld=lld
+	EOF
+
+	newins - gentoo-gcc-install.cfg <<-EOF
+		# This file is maintained by gcc-config.
+		# It is used to specify the selected GCC installation.
+	EOF
+
+	newins - default.cfg <<-EOF
+		# This configuration file is used by unprefixed clang invocations
+		# (e.g. clang, clang++, etc.).
+
+		@gentoo-runtimes.cfg
+		@gentoo-gcc-install.cfg
+	EOF
+
+	for abi in $(get_all_abis); do
+		local abi_chost=$(get_abi_CHOST "${abi}")
+		newins - "${abi_chost}.cfg" <<-EOF
+			# This configuration file is used by clang invocations prefixed
+			# ${abi_chost} (e.g. ${abi_chost}-clang).
+
+			@default.cfg
+		EOF
+	done
+}
+
+pkg_preinst() {
+	# TODO: move this to gcc-config and call it here?
```

Comment:
Add a bug reference in this comment & the commit to bug [868639](https://bugs.gentoo.org/show_bug.cgi?id=868639).